crates/sui-storage/src/mutex_table.rs

```diff
@@ -125,45 +132,50 @@ impl<K: Hash + std::cmp::Eq + Send + Sync + 'static> MutexTable<K> {
 
     pub async fn acquire_lock(&self, k: K) -> LockGuard {
         let lock_idx = self.get_lock_idx(&k);
-        let map = self.lock_table[lock_idx].read().await;
-        if let Some(element) = map.get(&k) {
+        let element = {
+            let map = self.lock_table[lock_idx].read().await;
+            map.get(&k).cloned()
+        };
+        if let Some(element) = element {
             LockGuard(element.clone().lock_owned().await)
         } else {
             // element doesn't exist
-            drop(map);
-            let mut map = self.lock_table[lock_idx].write().await;
-            let element = map.entry(k).or_insert_with(|| Arc::new(Mutex::new(())));
+            let element = {
+                let mut map = self.lock_table[lock_idx].write().await;
+                map.entry(k)
+                    .or_insert_with(|| Arc::new(Mutex::new(())))
+                    .clone()
+            };
             LockGuard(element.clone().lock_owned().await)
```

Comment:
Is the clone() here still needed?  You just cloned it above when retrieving element.

Reply:
Yeah, I am surprised linter did not catch this